blogging/views.py

```diff
@@ -0,0 +1,144 @@
+from django.shortcuts import render, redirect
+from django.http import HttpResponse, HttpResponseRedirect, Http404
+from django.template import loader
+from django.utils import timezone
+from blogging.models import Post, Category
+from blogging.forms import PostForm, PublishForm, ListForm
+
+
+def list_view(request):
+    """
+    Displays a list view of each published post.
+
+    In the real world I'd likely want to limit the number of posts displayed incase
+    the blog grows large, with a 'display more' at the bottom.
+    Don't know how to do that yet though...
+    """
+    if request.method == "POST":
+        if not request.user.is_authenticated:
+            return redirect("/login")
+
+        form = PublishForm(request.POST)
+
+        if form.is_valid():
+            post = Post.objects.get(id=form.cleaned_data["post_id"])
+
+            if request.user == post.author:
+                post.published_date = None
+                post.save()
+                return redirect("/")
+        else:
+            print(form.errors)
```

Comment:
I'd setup some proper logging instead of using print: https://docs.djangoproject.com/en/2.2/topics/logging/

Reply:
This was actually just a remnant of testing that should have been removed for production.  Thank you for pointing it out.  On a similar note, when intentionally leaving in things such as that, I've been having good luck with loguru.